src/envoy/mixer/grpc_transport.cc

```diff
@@ -72,6 +82,18 @@ bool GrpcTransport::IsMixerServerConfigured(Upstream::ClusterManager& cm) {
   return cm.get(kMixerServerClusterName) != nullptr;
 }
 
+void GrpcTransport::onPreRequestCustomizeHeaders(Http::HeaderMap& headers) {
+  if (!headers_) return;
+
+  for (const auto& key : kHttpTraceHeaders) {
+    const HeaderEntry* entry = headers_->get(key);
```

Comment:
It works.  I have tested. 

Apparently,  headers_ has everything.   
inline_headers_ is just an optimization to access the headers_